### PR TITLE
remove nodes not in json task #10

### PR DESCRIPTION
**Pull request by blackout314**
task #10 
remove nodes not in list

### CODE REVIEW

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) everybody who has role `REV` is banned at [#54](https://github.com/zold-io/zold.github.io/issues/54); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by blackout314** [comment]
@yegor256 yes i'm sure, because in javascript "Time" is "Object"
https://i.imgur.com/9sXzPHd.png

**Comment by blackout314** [comment]
i do another commit for better intent

File: js/map.js [hunk 1]
```diff
@@ -59,6 +61,19 @@ function refresh_list(host, map) {
...
     var remotes = data.all;
     console.log(remotes.length + ' remote nodes found at ' + host);
+
+    currentHashing = new Date().getTime();
+    $.each(remotes, function (i, r) {
+      var host = r.host, port = r.port;
+      var coords = host + ':' + port;
+      var items = $('#remotes-table tr[data-coords="' + coords + '"]');
+      var item = items.first();
+
```
Comment (yegor256) [inline js/map.js]: @blackout314 check this: https://www.yegor256.com/2014/11/03/empty-line-code-smell.html

File: js/map.js [hunk 2]
```diff
@@ -59,6 +61,19 @@ function refresh_list(host, map) {
...
+      item.attr('data-check', currentHashing);
+    });
+
```
Comment (yegor256) [inline js/map.js]: @blackout314 same here. Don't use empty lines, it's a code smell.

File: js/map.js [hunk 3]
```diff
@@ -59,6 +61,19 @@ function refresh_list(host, map) {
   $.getJSON('http://' + host + '/remotes', function(data) {
     var remotes = data.all;
     console.log(remotes.length + ' remote nodes found at ' + host);
+
+    currentHashing = new Date().getTime();
+    $.each(remotes, function (i, r) {
+      var host = r.host, port = r.port;
```
Comment (yegor256) [inline js/map.js]: @blackout314 what is the point of introducing these two vars?

File: js/map.js [hunk 4]
```diff
@@ -59,6 +61,19 @@ function refresh_list(host, map) {
   $.getJSON('http://' + host + '/remotes', function(data) {
     var remotes = data.all;
     console.log(remotes.length + ' remote nodes found at ' + host);
+
+    currentHashing = new Date().getTime();
```
Comment (yegor256) [inline js/map.js]: @blackout314 why here it's `getTime` and above it's just `new Date`?

**Comment by yegor256** [comment]
@blackout314 see above

File: js/map.js [hunk 5]
```diff
@@ -59,6 +61,13 @@ function refresh_list(host, map) {
   $.getJSON('http://' + host + '/remotes', function(data) {
     var remotes = data.all;
     console.log(remotes.length + ' remote nodes found at ' + host);
+    currentTimestamp = new Date().getTime();
```
Comment (yegor256) [inline js/map.js]: @blackout314 I can't understand what is the point of having this value in a global variable. You are not using it anywhere else, except this method. Why keeping this value globally?

**Comment by yegor256** [comment]
@blackout314 looks better now, see above

**Comment by blackout314** [comment]
@yegor256 you are right, global isnt useful in this case
my original idea is compare `timestamp[n-1]` with `timestamp[n]`

but better simple like now

File: js/map.js [hunk 6]
```diff
@@ -57,8 +57,15 @@ function refresh(host, map) {
 
 function refresh_list(host, map) {
   $.getJSON('http://' + host + '/remotes', function(data) {
-    var remotes = data.all;
+    var remotes = data.all,
```
Comment (yegor256) [inline js/map.js]: @blackout314 what is the point of this `remotes` variable? Can't we use `data.all` instead?

File: js/map.js [hunk 7]
```diff
@@ -68,32 +75,26 @@ function put_markers(map, remotes) {
     var host = r.host, port = r.port;
```
Comment (yegor256) [inline js/map.js]: @blackout314 same here, why these variables? Check this out: https://www.yegor256.com/2015/09/01/redundant-variables-are-evil.html

**Comment by yegor256** [comment]
@blackout314 see above. Variables are evil, try to have them as few as possible.

File: js/map.js (hunk 7 above)
Comment (blackout314) [reply js/map.js]: @yegor256 is original code, see annotation

File: js/map.js (hunk 6 above)
Comment (blackout314) [reply js/map.js]: @yegor256 is original code, not mine

**Comment by blackout314** [comment]
@yegor256 see history
https://github.com/zold-io/zold.github.io/blob/08fe64742b0771450fab07f77f20a3472034ec9f/js/map.js

**Comment by yegor256** [comment]
@rultor mertge

**Comment by 0crat** [comment]
Job `gh:zold-io/zold.github.io#54` is not assigned, can't get performer

**Comment by 0crat** [comment]
The job [#54](https://github.com/zold-io/zold.github.io/issues/54) is now out of scope

**Pull request merged**